### PR TITLE
[WIP] Fixes for 0.7

### PR DESCRIPTION
**Pull request by rened**
Addresses #316. Currently waiting for https://github.com/denizyuret/AutoGrad.jl/pull/68 and work in progress.

But had most of its unit tests pass a couple of weeks ago.

### CODE REVIEW

File: src/gpu.jl [hunk 1]
```diff
@@ -47,7 +47,7 @@ macro nvml(fun,x...)
     esc(Expr(:macrocall,Symbol("@cuda"),"nvml",fun,x...))
 end
 
-const Cptr = Ptr{Void}
+const Cptr = Ptr{Nothing}
```
Comment (GunnarFarneback) [inline src/gpu.jl]: This could also be a Cvoid (same in nvrtc.jl).

File: .gitignore [hunk 2]
```diff
@@ -33,3 +33,4 @@ setup.sh
 *.log
 *.prof
 *.gz
+dev
```
Comment (CarloLucibello) [inline .gitignore]: please add Manifest.toml here

**Comment by denizyuret** [comment]
The latest master should support Julia 0.7 and 1.0. Please test.

**Comment by rened** [comment]
Seems to work great so far! Thanks a lot!

**Pull request closed without merging**